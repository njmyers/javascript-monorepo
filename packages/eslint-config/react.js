module.exports = {
  parser: 'babel-eslint',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
  env: {
    'shared-node-browser': true,
  },
  extends: ['plugin:react/recommended'],
  plugins: ['react', 'react-hooks', 'jsx-a11y'],
  rules: {
    'jsx-a11y/label-has-associated-control': 'error',
    'react/display-name': 'off',
    'react/prop-types': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'error',
  },
  settings: {
    react: {
      version: 'detect',
    },
  },
};
