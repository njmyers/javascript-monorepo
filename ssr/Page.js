import React, { Component } from 'react';
import Size from '../src';

const LoremIpsum = ({ sizes } = {}) => {
  const style = {
    boxSizing: 'border-box',
    background: '#7f3535',
    width: '100%',
    padding: '30px',
  };

  console.log(sizes);

  return (
    <React.Fragment>
      <article style={style}>
        <h2>Sized Component</h2>
        <h3>Component</h3>
        <li>width: {sizes.component.width}</li>
        <li>height: {sizes.component.height}</li>
        <h3>Window</h3>
        <li>innerWidth: {sizes.window.innerWidth}</li>
        <li>innerHeight: {sizes.window.innerHeight}</li>
        <li>outerWidth: {sizes.window.outerWidth}</li>
        <li>outerHeight: {sizes.window.outerHeight}</li>
        <h3>Breakpoints</h3>
        <li>tablet: {sizes.tablet.toString()}</li>
        <li>laptop: {sizes.laptop.toString()}</li>
        <li>mobile: {sizes.mobile.toString()}</li>
        <li>orientation: {sizes.orientation}</li>
        <h3>Position</h3>
        <li>component is in view: {sizes.inView.toString()}</li>
        <p>
          <i>note: check the console for callbacks</i>
        </p>
      </article>
    </React.Fragment>
  );
};

// Define some custom size flags
const tablet = (node) => window.innerWidth < 1440 && window.innerWidth >= 768;
const laptop = (node) => window.innerWidth >= 1440;

// custom flags for our options object
const custom = [
  {
    name: 'tablet',
    fn: tablet,
    schema: '',
    subscriptions: {
      resize: true,
    },
  },
  {
    name: 'laptop',
    fn: laptop,
    schema: '',
    subscriptions: {
      resize: true,
    },
  },
];

// Lets enable all modes so we can test all of the code for SSR functionality.
const SizedLorem = Size({
  measureWindow: true,
  component: true,
  mobile: true,
  orientation: true,
  inView: true,
  custom,
})(LoremIpsum);

/**
 * Demo Component to test RSC with SSR
 */
class Page extends Component {
  constructor(props) {
    super(props);
    // just inline some basic styles so we don't have to compile any css with webpack
    this.article = {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'flex-start',
      padding: '0 5rem',
      color: 'white',
      background: '#65655c',
      minHeight: '100vh',
    };
  }
  // Testing CB function with SSR
  onSize = (sizes) => {
    console.log(sizes);
  };

  render() {
    return (
      <section style={this.article}>
        <h1>Server Side Rendering Demo</h1>
        <SizedLorem onSize={this.onSize} />
      </section>
    );
  }
}

export default Page;
