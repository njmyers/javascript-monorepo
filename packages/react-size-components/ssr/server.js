import path from 'path';
import express from 'express';
import React from 'react';
import { renderToString } from 'react-dom/server';
import Page from './Page';
import template from './template';

const port = 5000;
const server = express();

// Add our react build folder as a static path for express. Use path since this project is in a sub-folder.
server.use('/build', express.static(path.join(__dirname, 'build')));

// Simple demo so no need for complex rendering. Simply render the the react page as a string and send it in the express response.
server.get('/', (req, res) => {
  res.send(
    template({
      title: 'Server Side Rendering Demo',
      body: renderToString(<Page />),
    })
  );
});

server.listen(port, () => {
  console.log(`SSR (Server Side Rendering) demo running on port ${port}`);
});
