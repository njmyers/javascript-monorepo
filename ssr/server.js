import express from 'express';
import React from 'react';
import { renderToString } from 'react-dom/server';
import Page from './Page';
import template from './template';
import path from 'path';

const port = 3000;
const server = express();

server.use('/build', express.static(path.join(__dirname, 'build')));

server.get('/', (req, res) => {
    res.send(
        template({
            title: 'Test Page',
            body: renderToString(<Page />),
        })
    );
});

server.listen(port, () => {
    console.log(`SSR (Server Side Rendering) demo running on port ${port}`);
});
