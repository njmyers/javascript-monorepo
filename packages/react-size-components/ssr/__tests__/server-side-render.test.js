/**
 * @jest-environment node
 */

// Removes jest provided window global
// https://jestjs.io/docs/en/configuration.html#testenvironment-string

import * as React from 'react';
import { renderToString } from 'react-dom/server';
import Page from '../Page';

/** Should throw an error for accesing window before componentDidMount */
class WillThrow extends React.Component {
  render() {
    return (
      <article>
        <h1>Window Height</h1>
        <p>{window.innerHeight}</p>;
      </article>
    );
  }
}

describe('SSR functionality for size components', () => {
  test('sanity check, should throw error', () => {
    // accessing window in node
    expect(() => window).toThrowErrorMatchingSnapshot();
    // accessing window in lifecycle
    expect(() => renderToString(<WillThrow />)).toThrowErrorMatchingSnapshot();
  });

  test('it renders to string without errors', () => {
    // accessing window in proper lifecycle
    expect(renderToString(<Page />)).toMatchSnapshot();
  });
});
