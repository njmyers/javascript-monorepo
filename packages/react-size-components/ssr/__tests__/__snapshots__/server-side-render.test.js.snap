// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR functionality for size components it renders to string without errors 1`] = `"<section style=\\"display:flex;flex-direction:column;justify-content:center;align-items:flex-start;padding:0 5rem;color:white;background:#65655c;min-height:100vh\\" data-reactroot=\\"\\"><h1>Server Side Rendering Demo</h1><article style=\\"box-sizing:border-box;background:#7f3535;width:100%;padding:30px\\"><h2>Sized Component</h2><h3>Component</h3><li>width: <!-- -->0</li><li>height: <!-- -->0</li><h3>Window</h3><li>innerWidth: <!-- -->0</li><li>innerHeight: <!-- -->0</li><li>outerWidth: <!-- -->0</li><li>outerHeight: <!-- -->0</li><h3>Breakpoints</h3><li>tablet: </li><li>laptop: </li><li>mobile: <!-- -->false</li><li>orientation: <!-- -->landscape</li><h3>Position</h3><li>component is in view: <!-- -->false</li><p><i>note: check the console for callbacks</i></p></article></section>"`;

exports[`SSR functionality for size components sanity check, should throw error 1`] = `"window is not defined"`;

exports[`SSR functionality for size components sanity check, should throw error 2`] = `"window is not defined"`;
